src/Middleware/Spa/SpaProxy/src/SpaProxyLaunchManager.cs

```diff
@@ -203,38 +221,86 @@ private void LaunchDevelopmentProxy()
 
 try
 {{
-  taskkill /T /F /PID {_spaProcess.Id};
+  taskkill /T /F /PID {_spaProcess!.Id};
 }}
 catch
 {{
 }}";
-                    var stopScriptInfo = new ProcessStartInfo(
-                        "powershell.exe",
-                        string.Join(" ", "-NoProfile", "-C", stopScript))
-                    {
-                        CreateNoWindow = true,
-                        WorkingDirectory = Path.Combine(AppContext.BaseDirectory, _options.WorkingDirectory)
-                    };
+            var stopScriptInfo = new ProcessStartInfo(
+                "powershell.exe",
+                string.Join(" ", "-NoProfile", "-C", stopScript))
+            {
+                CreateNoWindow = true,
+                WorkingDirectory = Path.Combine(AppContext.BaseDirectory, _options.WorkingDirectory)
+            };
 
-                    var stopProcess = Process.Start(stopScriptInfo);
-                    if (stopProcess == null || stopProcess.HasExited)
-                    {
-                        _logger.LogWarning($"SPA process shutdown script '{stopProcess?.Id}' failed to start. The SPA proxy might" +
-                            $" remain open if the dotnet process is terminated abruptly. Use the operating system command to kill" +
-                            $"the process tree for {_spaProcess.Id}");
-                    }
-                    else
-                    {
-                        _logger.LogDebug($"Watch process '{stopProcess}' started.");
-                    }
-                }
+            var stopProcess = Process.Start(stopScriptInfo);
+            if (stopProcess == null || stopProcess.HasExited)
+            {
+                _logger.LogWarning($"The SPA process shutdown script '{stopProcess?.Id}' failed to start. The SPA proxy might" +
+                    $" remain open if the dotnet process is terminated ungracefully. Use the operating system commands to kill" +
+                    $" the process tree for {_spaProcess!.Id}");
             }
-            catch (Exception exception)
+            else
             {
-                _logger.LogError(exception, $"Failed to launch the SPA development server '{_options.LaunchCommand}'.");
+                _logger.LogDebug($"Watch process '{stopProcess}' started.");
             }
         }
 
+        private void LaunchStopScriptMacOS()
+        {
+            var fileName = Guid.NewGuid().ToString("N") + ".sh";
+            var scriptPath = Path.Combine(AppContext.BaseDirectory, fileName);
```

Comment:
Why write it into the app directory? Could it go into a temp directory instead?

Reason: if for some reason this fails or takes a nontrivial amount of time to run, it will show up as a file in source control. Or if there is some file watcher (e.g., hot reload), it's going to trigger some action.

Reply:
AppContext.BaseDirectory points to the bin\Debug\net6.0 folder

Reply:
Ok, sounds fine then. I’m not certain how spaces in the path would be handled on macOS but assuming the script will still get executed in this case, all good!